### DISCUSSION TITLE
Changing ldap authentication from sAMAccountName to userPrincipalName

### DISCUSSION THREAD

**Discussion by deadsik** [authelia/authelia#6529] [Q&A] [answered]
Hello,

I am currently using ldap authentication using sAMAccountName.
There was a need to change the authentication type by userPrincipalName.

**For this purpose I added the following attribute to the configuration:**
`username_attribute: UserPrincipalName`

Authentication passes, but there is no access to the required page (secure.example.com), the following message is observed in the logs:
```
ip=192.168.0.1 stack="github.com/authelia/authelia/internal/handlers/handler_verify.go:458     VerifyGet.func1\ngithub.com/authelia/authelia/internal/middlewares/authelia_context.go:50 AutheliaMiddleware.func1.1\ngithub.com/fasthttp/router@v1.3.12/router.go:414                         (*Router).Handler\ngithub.com/authelia/authelia/internal/middlewares/log_request.go:14      LogRequestMiddleware.func1\ngithub.com/valyala/fasthttp@v1.24.0/server.go:2219                       (*Server).serveConn\ngithub.com/valyala/fasthttp@v1.24.0/workerpool.go:223                    (*workerPool).workerFunc\ngithub.com/valyala/fasthttp@v1.24.0/workerpool.go:195                    (*workerPool).getCh.func1\nruntime/asm_amd64.s:1371                                                 goexit"
time="2024-01-11T15:35:21Z" level=info msg="Access to https://secure.example.com/ (method GET) is not authorized to user <anonymous>, redirecting to https://login.example.com/?rd=https%3A%2F%2Fsecure.example.com%2F&rm=GET" method=GET path=/api/verify remote_ip=192.168.0.1
```

Perhaps someone has encountered a similar problem and needs to change something else?

**Comment by james-d-elliott** [comment, marked as answer]
This is very unlikely to be an issue with your LDAP config, the more likely reasons this usually happens are:

- The user is NOT authenticated.
- The users session has expired.
- The browser is not sending the session cookie due to:
  - The resource not being served via the `https://` scheme in the browser.
  - An extension preventing the cookie.
  - The cookie domain (set in the session configuration) is not valid for the site with issues.

**Reply by Namelessinterester** [reply]
Hi. To be more exact when we have default configuration username_attribute not specified(sAMAccountName is set). All functionality was working fine, like network rules DUO and etc.
But when just one parametr changed username_attribute: UserPrincipalName

First thing we have seen is that DUO stoped working (with such error time="2024-01-17T15:29:01Z" level=warning msg="Duo Push Auth failed to process the auth request for  from 10.32.44.28: Username is required. (username), error code 40002. This error often occurs if you've not setup the username in the Admin Dashboard." method=POST path=/api/secondfactor/duo remote_ip=10.32.44.28)

As you can see it says username is not set but just before adding username_attribute: UserPrincipalName it was fineю

**Reply by james-d-elliott** [reply]
Yeah DUO will stop working because the username changed like you wanted it to, that's to be expected. You have to fix that on the DUO end. This is exactly the same for any other thing that relies on the username, it must be updated to the new value.

**Reply by nightah** [reply]
You can also set aliases in Duo as opposed to changing the actual username. This is useful when supporting more than just a single field for login (username/email/etc).

**Reply by Namelessinterester** [reply]
One more question if I change username_attribute is headers somehow changed?
Cause what I did now, just disabled 2fa. And tried to login in to application (in my case it is karma) but so far it seems like not working. Like Remote-User header is missing....

**Reply by james-d-elliott** [reply]
The Remote-User response header will always be included at the authorization endpoints and can be copied with your given proxy. Changing the attribute will just change what value is displayed, not the existence of it.
